Document zip output feature

flomnes opened the pull request.

flomnes committed:
Document zip output feature

flomnes committed:
Make absence of pytest or numpy fatal at configure time (#850)

* [CMake] Make absence of pytest or numpy fatal at configure time

* Remove useless CMake parameter

* Use pip instead of pip3

* Do not setup python 3.x

* Use py 3.10

* Disable registry search in Python3

See https://cmake.org/cmake/help/latest/module/FindPython3.html

* Revert "Disable registry search in Python3"

This reverts commit 7e8a67d711458b1c6893ec6cd6f9f17f6f67e389.

* Remove Python 3.10_x86

* Python3_FIND_REGISTRY "LAST"

* Keep Python 3.10_x86

* Cache pip deps

* Fix windows-vcpkg.yml

* [Win] Also run tests on releases

* Remove 'pip' caching

* Rename step

flomnes committed:
Add Antares logo in solver logs (#861)

* Add Antares logo in solver logs

* Add const qualifier, enlarge logo

flomnes committed:
Output variables: remove useless methods for renewable clusters (#869)

Methods

- yearEndBuildPrepareDataForEachRenewableCluster
- yearEndResetRenewable

These methods do not produce any result, and sometimes produce segfaults.

payetvin committed:
[DEV] Refactored Layers by adding a LayerData class (#866)

* [REFAC] Added class LayerData in study.h

* [REFAC] Fix class members init order to avoid warnings

* [REFAC] Add comment in study contructor

* [REFAC] Move class LayerData to layerdata.c/h

* [REFAC] Fixing concatenate namespaces

* [REFAC] Fixed sonarcloud code smells

* [REFAC] Remove useless private: in study.h

* [REFAC] Use structured bindings

flomnes committed:
Remove unused "shedding strategy" (#788)

* Remove unused "shedding strategy"

* Remove I/O, move read to legacy

* Fix build

* Remove braces

flomnes committed:
Simplify day ahead reserve condition (#777)

* [OPT] Day ahead reserve condition

* Simplify bounds

flomnes committed:
Add changelog for 8.3.2, version++

payetvin committed:
Add OR-Tools+XPRESS to Ubuntu CI and merge files for centos CI (#859)

* ll

* Started to merge centos release and system yml

* [DEV CI] Merged centos release and system in centos.yml

* [DEV CI] Try fix docker_publish error

* [DEV CI] Try fix docker_publish and actions/checkout

* [DEV CI] Try fix Build xpress ortools: get release

* [DEV CI] Fixed if syntax

* [DEV CI] Fixed syntax error line 150

* [DEV CI] Fixed env.IS_RELEASE

* [DEV CI] Fixed env.IS_PUSH

* [DEV CI] Fixed fetch-depth in docker_publish

* [DEV CI] Fixed env var syntax

* [DEV CI] Fixed env var contexts

* [DEV CI] Fixed contexts conditions

* [DEV CI] Fix quote around true statement

* [DEV CI] Fix push/release invert

* [DEV CI] Remove souce /opt/rh/rh-git227/enable from ubuntu-system.yml

* [DEV CI] Deleted centos7 release and system yml files

* [DEV CI] Merged ubuntu system and release first try

* [DEV CI] Add jobs

* [DEV CI] Comment unfeasibility tests and deactivate other CI

* [DEV CI] Comment end to end tests

* [DEV CI] Comment test infinity

* [DEV CI] Fixed flomnes requests

* [DEV CI] Fixed submodules init

* [DEV CI] Added submodule init

* [DEV CI] Added debug in init

* [DEV CI] Try debug unfeasible tests

* [DEV CI] More debug in init

* [DEV CI] fix debug path in init

* [DEV CI] add debug path in init

* [DEV CI] Add debug in xpress download

* [DEV CI] Fix URL for or-tools download

* [DEV CI] Add var to store or-tools url

* [DEV CI] FIx var url or-tools

* [DEV CI] Deleted ubuntu-system/release, reactivate windows and centos CI

* [DEV CI] Add or-tools url in env var

* [DEV CI] Add or-tools url static for centos CI

* [DEV CI] Fix if condition

* [REFAC] Fix Florian comments

* [REFAC] Add ccache to sonarcloud

* [DEV CI] Add ccache log in ubuntu ci

* [DEV CI] Add ccache in centos + logs

* [DEV CI] Add ccache in centos

* [DEV CI] Upgrade ccache-action to 1.2

* [DEV CI] Add ccache to windows

* [DEV CI] Fix windows ccache

* [DEV CI] Add ccache to centos7 docker img

* [DEV CI] Use sccache for windows

* [DEV CI] Try sccache for windows

* [DEV CI] Try sccache flags for windows

* [DEV CI] Remove BUILD_OUTPUT_TEST in cmake config

* Ubuntu : formatting

* Ubuntu : same configure for push & release

* Ubuntu : fix path for antares-deps install dir

* Fix release asset names for Ubuntu

* Download OR-Tools in dedicated action

* Add missing mkdir

* Fix windows-vcpkg.yml

* Fix windows-release.yml

* Fix build for Ubuntu, Sonar

* Fix build for Ubuntu, Sonar (bis)

* Fix CMAKE_PREFIX_PATH

* Fix path

* [Win] Escape \ in path

Co-authored-by: PAYET Vincent Ext <payetvin@gm0winl513.bureau.si.interne>
Co-authored-by: Florian OMNES <florian.omnes@rte-france.com>

payetvin committed:
Add OR-Tools+XPRESS to Ubuntu CI and merge files for centos CI (#859)

* ll

* Started to merge centos release and system yml

* [DEV CI] Merged centos release and system in centos.yml

* [DEV CI] Try fix docker_publish error

* [DEV CI] Try fix docker_publish and actions/checkout

* [DEV CI] Try fix Build xpress ortools: get release

* [DEV CI] Fixed if syntax

* [DEV CI] Fixed syntax error line 150

* [DEV CI] Fixed env.IS_RELEASE

* [DEV CI] Fixed env.IS_PUSH

* [DEV CI] Fixed fetch-depth in docker_publish

* [DEV CI] Fixed env var syntax

* [DEV CI] Fixed env var contexts

* [DEV CI] Fixed contexts conditions

* [DEV CI] Fix quote around true statement

* [DEV CI] Fix push/release invert

* [DEV CI] Remove souce /opt/rh/rh-git227/enable from ubuntu-system.yml

* [DEV CI] Deleted centos7 release and system yml files

* [DEV CI] Merged ubuntu system and release first try

* [DEV CI] Add jobs

* [DEV CI] Comment unfeasibility tests and deactivate other CI

* [DEV CI] Comment end to end tests

* [DEV CI] Comment test infinity

* [DEV CI] Fixed flomnes requests

* [DEV CI] Fixed submodules init

* [DEV CI] Added submodule init

* [DEV CI] Added debug in init

* [DEV CI] Try debug unfeasible tests

* [DEV CI] More debug in init

* [DEV CI] fix debug path in init

* [DEV CI] add debug path in init

* [DEV CI] Add debug in xpress download

* [DEV CI] Fix URL for or-tools download

* [DEV CI] Add var to store or-tools url

* [DEV CI] FIx var url or-tools

* [DEV CI] Deleted ubuntu-system/release, reactivate windows and centos CI

* [DEV CI] Add or-tools url in env var

* [DEV CI] Add or-tools url static for centos CI

* [DEV CI] Fix if condition

* [REFAC] Fix Florian comments

* [REFAC] Add ccache to sonarcloud

* [DEV CI] Add ccache log in ubuntu ci

* [DEV CI] Add ccache in centos + logs

* [DEV CI] Add ccache in centos

* [DEV CI] Upgrade ccache-action to 1.2

* [DEV CI] Add ccache to windows

* [DEV CI] Fix windows ccache

* [DEV CI] Add ccache to centos7 docker img

* [DEV CI] Use sccache for windows

* [DEV CI] Try sccache for windows

* [DEV CI] Try sccache flags for windows

* [DEV CI] Remove BUILD_OUTPUT_TEST in cmake config

* Ubuntu : formatting

* Ubuntu : same configure for push & release

* Ubuntu : fix path for antares-deps install dir

* Fix release asset names for Ubuntu

* Download OR-Tools in dedicated action

* Add missing mkdir

* Fix windows-vcpkg.yml

* Fix windows-release.yml

* Fix build for Ubuntu, Sonar

* Fix build for Ubuntu, Sonar (bis)

* Fix CMAKE_PREFIX_PATH

* Fix path

* [Win] Escape \ in path

Co-authored-by: PAYET Vincent Ext <payetvin@gm0winl513.bureau.si.interne>
Co-authored-by: Florian OMNES <florian.omnes@rte-france.com>

flomnes committed:
Fix Windows CI (#879)

* [CI] Fix CMAKE_PREFIX_PATH for windows

* Remove ORTOOLS_DIR

flomnes committed:
Fix Windows CI (#879)

* [CI] Fix CMAKE_PREFIX_PATH for windows

* Remove ORTOOLS_DIR

flomnes committed:
[DOC] OR-Tools build

flomnes committed:
Replace "Flat" view with "By area" view + refactor (#755)

* UI: Change and rename "upstream/downstream" view

Instead of
A
  Z

Display both
A
  Z

Z
  A

So that the user can see what areas are linked to any area.

* [skip ci][ui] Remove flat view

* Rename areaLinks -> areaToListOfLinks

* [sonar] Fix code smells

* [GUI] Replace "Flat" view with "By area" view

* PR remarks + cleaning

* Add link count

* PR remarks

* Fix code smells

* Code smells

* Fix code smell

flomnes committed:
Replace "Flat" view with "By area" view + refactor (#755)

* UI: Change and rename "upstream/downstream" view

Instead of
A
  Z

Display both
A
  Z

Z
  A

So that the user can see what areas are linked to any area.

* [skip ci][ui] Remove flat view

* Rename areaLinks -> areaToListOfLinks

* [sonar] Fix code smells

* [GUI] Replace "Flat" view with "By area" view

* PR remarks + cleaning

* Add link count

* PR remarks

* Fix code smells

* Code smells

* Fix code smell

guilpier-code commented on docs/reference-guide/13-file-format.md:
```diff
@@ -1,6 +1,13 @@
 
+## v8.4.0
+### Input
+In file **settings/generaldata.ini**, in existing section `output`, add property `result-format` [str]. Default value = `files`. If this property is set to `zip`, all results are written into a single zip archive, instead of multiple files.
```
Are you sure that the value **files** for item **result-format** is a good value name ?
Would **txt-files** not be better ? Or **full-txt-format** ?

flomnes committed:
Fix warning in hydro.ini loading function (#883)

* Fix warning in hydro.ini loading function

* [FIX] Hydro warning container

Co-authored-by: Vincent Payet <vincent.payet@rte-france.com>

flomnes committed:
Fix warning in hydro.ini loading function (#883)

* Fix warning in hydro.ini loading function

* [FIX] Hydro warning container

Co-authored-by: Vincent Payet <vincent.payet@rte-france.com>

flomnes committed:
Forbid OR-Tools + Split MPS (#877)

* Forbid Split MPS + OR-Tools

* Fix build

flomnes committed:
PR remarks

flomnes committed:
Use average for variable MARG. COST (links) (#881)

* Use average for variable MARG. COST (links)

* Revert "Use average for variable MARG. COST (links)"

This reverts commit bafe8018fc83bc9250c08df0d6f6becde53dd8a6.

* Use average for MARG. COST (links)

flomnes committed:
Use average for variable MARG. COST (links) (#881)

* Use average for variable MARG. COST (links)

* Revert "Use average for variable MARG. COST (links)"

This reverts commit bafe8018fc83bc9250c08df0d6f6becde53dd8a6.

* Use average for MARG. COST (links)

flomnes committed:
Merge remote-tracking branch 'github/feature/thread-writer-zip' into doc/zip-output

flomnes committed:
13-file-format.md

flomnes committed:
Allow more than 100 thermal clusters (#886)

flomnes committed:
Allow more than 100 thermal clusters (#886)

guilpier-code committed:
MPS writer refactoring : defining an mps writer interface and consequences (#863)

* mps writer refactoring : defining an mps writer interface and consequences

Each way of printing MPS files (full historic, or-tools way, split format or none) is now isolated
inside a different class. Each of these ways can be removed or changed without impacting the client code.
A factory function contains the logic that leads to the creation of a particular MPS writer.

* mps writer refactoring : fix compilation issue on Ubuntu

* [skip ci] mps writer refactoring : add a simple comment

* mps writer refactoring : fix a crash on tests

* mps writer refactoring : enhancing the MPS writer factory (from functions to a class)

* [skip ci] mps writer refactoring : remove commented coded

* MPS files printing : restore print of 1st and 2nd optimization, whether split or not   (#872)

* mps dumping 1st and 2nd optim : just a save (does not compile)

* mps dumping 1st and 2nd optim : end of functionality development (see following comments)

- GUI : the MPS export status has now 4 items, the user can choose one of them (no mps, mps for optim 1 or optim 2, or both).
- consequences in the solver

* [skip ci] mps dumping 1st and 2nd optim : cleaning + renaming

* mps dumping 1st and 2nd optim : take mps files new names into account in infeasible problem tests

* Make absence of pytest or numpy fatal at configure time (#850)

* [CMake] Make absence of pytest or numpy fatal at configure time

* Remove useless CMake parameter

* Use pip instead of pip3

* Do not setup python 3.x

* Use py 3.10

* Disable registry search in Python3

See https://cmake.org/cmake/help/latest/module/FindPython3.html

* Revert "Disable registry search in Python3"

This reverts commit 7e8a67d711458b1c6893ec6cd6f9f17f6f67e389.

* Remove Python 3.10_x86

* Python3_FIND_REGISTRY "LAST"

* Keep Python 3.10_x86

* Cache pip deps

* Fix windows-vcpkg.yml

* [Win] Also run tests on releases

* Remove 'pip' caching

* Rename step

* mps dumping 1st and 2nd optim : correction due to review

* mps dumping 1st and 2nd optim : making compilation pass on Linux

* mps dumping 1st and 2nd optim : corrections due to sonarQ review

* [ski ci] mps dumping 1st and 2nd optim : documentation

* mps dumping 1st and 2nd optim : corrections due to review

Co-authored-by: Florian Omnès <florian.omnes@rte-france.com>

* Rewrite mpsWriterFactory::doWeExportMPS (#888)

Co-authored-by: Florian Omnès <florian.omnes@rte-france.com>

guilpier-code committed:
MPS writer refactoring : defining an mps writer interface and consequences (#863)

* mps writer refactoring : defining an mps writer interface and consequences

Each way of printing MPS files (full historic, or-tools way, split format or none) is now isolated
inside a different class. Each of these ways can be removed or changed without impacting the client code.
A factory function contains the logic that leads to the creation of a particular MPS writer.

* mps writer refactoring : fix compilation issue on Ubuntu

* [skip ci] mps writer refactoring : add a simple comment

* mps writer refactoring : fix a crash on tests

* mps writer refactoring : enhancing the MPS writer factory (from functions to a class)

* [skip ci] mps writer refactoring : remove commented coded

* MPS files printing : restore print of 1st and 2nd optimization, whether split or not   (#872)

* mps dumping 1st and 2nd optim : just a save (does not compile)

* mps dumping 1st and 2nd optim : end of functionality development (see following comments)

- GUI : the MPS export status has now 4 items, the user can choose one of them (no mps, mps for optim 1 or optim 2, or both).
- consequences in the solver

* [skip ci] mps dumping 1st and 2nd optim : cleaning + renaming

* mps dumping 1st and 2nd optim : take mps files new names into account in infeasible problem tests

* Make absence of pytest or numpy fatal at configure time (#850)

* [CMake] Make absence of pytest or numpy fatal at configure time

* Remove useless CMake parameter

* Use pip instead of pip3

* Do not setup python 3.x

* Use py 3.10

* Disable registry search in Python3

See https://cmake.org/cmake/help/latest/module/FindPython3.html

* Revert "Disable registry search in Python3"

This reverts commit 7e8a67d711458b1c6893ec6cd6f9f17f6f67e389.

* Remove Python 3.10_x86

* Python3_FIND_REGISTRY "LAST"

* Keep Python 3.10_x86

* Cache pip deps

* Fix windows-vcpkg.yml

* [Win] Also run tests on releases

* Remove 'pip' caching

* Rename step

* mps dumping 1st and 2nd optim : correction due to review

* mps dumping 1st and 2nd optim : making compilation pass on Linux

* mps dumping 1st and 2nd optim : corrections due to sonarQ review

* [ski ci] mps dumping 1st and 2nd optim : documentation

* mps dumping 1st and 2nd optim : corrections due to review

Co-authored-by: Florian Omnès <florian.omnes@rte-france.com>

* Rewrite mpsWriterFactory::doWeExportMPS (#888)

Co-authored-by: Florian Omnès <florian.omnes@rte-france.com>

flomnes committed:
Add "Output aggregation change" in changelog

flomnes committed:
Add entries to the changelog

flomnes committed:
Add 'continue-on-error' flags

flomnes committed:
Merge remote-tracking branch 'github/develop' into release/8.3.2

flomnes committed:
Use different ccache keys for OR-Tools=ON/OR-Tools=OFF

flomnes committed:
Merge pull request #875 from AntaresSimulatorTeam/release/8.3.2

Release 8.3.2

flomnes committed:
RC=1

flomnes committed:
Fix windows-release.yml (syntax)

flomnes committed:
Fix sonarcloud.yml (syntax)

flomnes committed:
Fix sonarcloud.yml (syntax)

flomnes committed:
Ubuntu : always use cache

flomnes committed:
Fix windows-release.yml (missing pip packages)

flomnes committed:
RC=3

flomnes committed:
Ubuntu: fix artifact name

flomnes committed:
RC=4

flomnes committed:
Set RC=5

flomnes committed:
One single job publishes assets

flomnes committed:
[TMP][Ubuntu] Remove tests

flomnes committed:
Fix indentation

flomnes committed:
Add missing dl

flomnes committed:
Set RC=6

flomnes committed:
Try Ubuntu

flomnes committed:
Revert "Try Ubuntu"

This reverts commit f70ea5701adc21338bddba441297cd713d9cf865.

flomnes committed:
Ubuntu: publish all assets

flomnes committed:
Use alexellis/upload-assets@0.2.2

flomnes committed:
Add token

flomnes committed:
RC=7

flomnes committed:
Rename .tar.gz / .deb

flomnes committed:
RC=8

flomnes committed:
Add missing cd

flomnes committed:
Add missing quote

flomnes committed:
Don't fail on mv same file

flomnes committed:
Fix missing label

flomnes committed:
Upload assets in a single job

flomnes committed:
Windows release : disable tests

flomnes committed:
RC=14

flomnes committed:
Bump SimTest v8.3.2

flomnes committed:
Finalize changelog

flomnes committed:
Fix 13-file-format.md

flomnes committed:
Revert "[TMP][Ubuntu] Remove tests"

This reverts commit 665b8bbdd6363d87ea2f0d9f31f1a964ff8ae8e9.

flomnes committed:
Revert "Windows release : disable tests"

This reverts commit 9f82b21d11bffbb17e3583d1af971e0bb115ea48.

flomnes committed:
continue-on-error: false for short-tests

flomnes committed:
RC=0

flomnes committed:
Fix section name in 13-file-format.md

flomnes committed:
Merge pull request #880 from AntaresSimulatorTeam/doc/or-tools-build

flomnes committed:
Merge remote-tracking branch 'github/develop' into doc/zip-output

flomnes committed:
Add missing section
